### PR TITLE
Enforce graph visualisation, add T-FDP layout, improve UI/UX

### PR DESCRIPTION
**Pull request by qrutyy**
### Main goal - improve the graph representation for better UX.

Thats why I managed to:

- Implement _T-FDP_ layout, that will be customisable by user in future)
- Implement _'add Edge'_ and _'add Vertex'_ methods for user. (yeah, we love them)
- Improve the intuitiveness of the UI by adding _'FAQ'_ button, that displays the type of graph and its stored data type.
- Improve the UX (the more the better) by adding ability to drag the graph and zoom it.
- Display vertices ID
- Correct connection between _View_ and _ViewModel_, so _GraphView_ updates after some changes were made.
- Clean up the file structure

**TODO:** 
- mb optimize layout
- sql connection
- integration tests
- add two other tabs (_FileControl_, _Analyze_)
- make theme switchable

### CODE REVIEW

File: app/src/main/kotlin/view/graph/GraphView.kt [hunk 1]
```diff
@@ -1,17 +1,63 @@
...
+        if (updateRequired.value.value) {
+            viewModel.applyForceDirectedLayout(740.0, 650.0)
+        }
+        viewModel.verticesVM.forEach { v -> VertexView(v, currentScaleState.value) }
+        viewModel.edgesVM.forEach { e -> EdgeView(e, currentScaleState.value) }
+
+        viewModel.updateIsRequired.value = false
     }
-}
+}
```
Comment (shakareem) [inline app/src/main/kotlin/view/graph/GraphView.kt]: No new line in the end of file

File: app/src/main/kotlin/view/utils/FAQBox.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                    }
+
+                    is HoverInteraction.Exit -> {
+                        isHovered = false
+                    }
+                }
+            }
+        }
+    }
+}
```
Comment (shakareem) [inline app/src/main/kotlin/view/utils/FAQBox.kt]: And here

File: app/src/main/kotlin/view/utils/ZoomBox.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            }
+            Spacer(modifier = Modifier.height(8.dp))
+            FloatingActionButton(onClick = {
+                currentScale.value = (currentScale.value / 1.1f).coerceIn(0.7f, 1.9f)
+            }) {
+                Text("-")
+            }
+        }
+    }
+}
```
Comment (shakareem) [inline app/src/main/kotlin/view/utils/ZoomBox.kt]: And here

File: app/src/main/kotlin/viewmodel/MainScreenViewModel.kt [hunk 4]
```diff
@@ -1,20 +1,28 @@
...
+    val graphViewModel =
+        GraphViewModel(graph, showVerticesIds, showVerticesData, graphType, setDirectionState(currentGraphType))
 
-    init { // here will be a placement-function call
-        TestRepresentation().place(740.0, 650.0, graphViewModel.verticesVM)
+    init {
+        TFDPLayout().place(740.0, 650.0, graphViewModel.verticesVM)
     }
 
     //    fun setEdgeColor
```
Comment (shakareem) [inline app/src/main/kotlin/viewmodel/MainScreenViewModel.kt]: mb remove?

File: app/src/main/kotlin/viewmodel/graph/TFDPLayout.kt [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            k++
+        }
+
+        for (vi in vertices) { // check borders
+            vi.x.value = vi.x.value.coerceIn(0.dp, (width.toFloat() - 360 - vi.radius.value * 2).dp)
+            vi.y.value = vi.y.value.coerceIn(0.dp, (height.toFloat() - vi.radius.value * 2).dp)
+            println("${vi.x.value} ${vi.y.value}")
+        }
+    }
+}
```
Comment (shakareem) [inline app/src/main/kotlin/viewmodel/graph/TFDPLayout.kt]: And here

**Review by shakareem — CHANGES_REQUESTED** [review]
Refactor a lil

**Comment by qrutyy** [comment]
> Refactor a lil

Done

**Review by shakareem — APPROVED** [review]

**Review by spisladqo — APPROVED** [review]
So good and so helpful!

**Pull request merged**